wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -291,28 +333,59 @@ func (v *Wrapper) getKeyVaultClient(withCertPool *x509.CertPool) (*azkeys.Client
 	var err error
 	var cred azcore.TokenCredential
 
-	switch {
-	// Use a fully specified tenant, client, secret if provided
-	case v.tenantID != "" && v.clientID != "" && v.clientSecret != "":
-		cred, err = azidentity.NewClientSecretCredential(v.tenantID, v.clientID, v.clientSecret, nil)
+	switch v.authMethod {
```

Comment:
Could we refactor this one into a method and error on `None` as a value, and then in this parent method, if `None` is set, use the `switch`/`case` conditionals in the `None` branch below to re-dispatch this one with named methods?

And perhaps rename `None` to `Automatic` or something? 

Then I think it becomes more explicit what is happening; credential acquisition happens in a helper method with explicit single-branch flow and this parent method either directly acquires it or, in the case of the default `Automatic` selection, we explicitly call acquisition accordingly. 

Thoughts?

Reply:
sound reasonable to me, I will also add certificate base authentication, as this is the Microsoft recommended way of interacting with Azure service from outside Azure managed environments

Reply:
just pushed a few improvements, including support for certificate authentication)